src/components/structures/RoomView.tsx

```diff
@@ -528,7 +530,18 @@ export default class RoomView extends React.Component<IProps, IState> {
     }
 
     shouldComponentUpdate(nextProps, nextState) {
-        return (objectHasDiff(this.props, nextProps) || objectHasDiff(this.state, nextState));
+        const hasPropsDiff = objectHasDiff(this.props, nextProps);
+
+        const newUpgradeRecommendation = nextState.upgradeRecommendation || {}
```

Comment:
A comment for why we're explicitly pulling out this variable would be appreciated